tensorflow/python/data/ops/readers.py

```diff
@@ -84,14 +86,15 @@ def __init__(self, filenames, compression_type=None, buffer_size=None):
     """Creates a `TFRecordDataset`.
 
     Args:
-      filenames: A `tf.string` tensor containing one or more filenames.
+      filenames: A `tf.string` tensor containing one or more filenames or `PathLike` objects.
       compression_type: (Optional.) A `tf.string` scalar evaluating to one of
         `""` (no compression), `"ZLIB"`, or `"GZIP"`.
       buffer_size: (Optional.) A `tf.int64` scalar representing the number of
         bytes in the read buffer. 0 means no buffering.
     """
     super(_TFRecordDataset, self).__init__()
     # Force the type to string even if filenames is an empty list.
+    filenames = compat.path_to_str(filenames)
```

Comment:
Does this work if `filenames` is a `tf.Tensor` or iterable of `tf.Tensor` objects?

Reply:
@mrry just test locally and it does not. Is there an example you know of in the code base with a 'TensorFlow' standard way of handling this for a best practice implementation? i.e. to differentiate a tensor vs native python iterable etc?

Otherwise could put a check for `and not isinstance(p, tf.Tensor))` when checking for iterability

Reply:
@mrry updated with a check for `ops.Tensor` as well

Reply:
I think the new logic should be fine, but can you please add a test that uses `PathLike`?